### PR TITLE
Setting customized sdk url and sdk resource as non customized ignores the sdk resource value #2097

### PR DESCRIPTION
**Pull request by YeeSkywalker**
Sign out: Yee Lin
Date: Jan 22nd, 2022
Issue: #2097

### CODE REVIEW

**Comment by YeeSkywalker** [comment]
Cannot pass Grinder tests if set `SDK_URL = nightly` and `CUSTOMIZED_SDK_URL` is not empty. Here is the link to Grinder test: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/3188/console

**Comment by smlambert** [comment]
We need to clarify what behaviours we expect to happen in this code:

|SDK_RESOURCE value | CUSTOMIZED_SDK_URL value | JDK_IMPL | Expected behaviour |
| --------- | ------------ | ---------- | ------- |
| nightly | "" | hotspot | Fetch the latest nightly EA binary from the adoptium API |
| nightly | "" | openj9 | Fetch the latest nightly binary from the adoptium API - this is a very old build, we should probably just print a warning |
| nightly | Top-level link to OpenJ9 artifactory | openj9 | Fetch nightly build from OpenJ9 artifactory |
| releases | "" | hotspot | Fetch the latest release GA binary from the adoptium API |
| releases | "" | openj9 |  |
| customized | "" | hotspot or openj9 | Print the newly added error added in this PR |
| customized | A link to a binary ending in tar.gz or zip  | hotspot or openj9 | Try to fetch the binary defined in the CUSTOMIZED_SDK_URL |

**Comment by YeeSkywalker** [comment]
| SDK_RESOURCE value  | CUSTOMIZED_SDK_URL value | JDK_IMPL | Grinder to verify |
| ------------- | ------------- |
| Content Cell  | Content Cell  |
| Content Cell  | Content Cell  |

**Comment by smlambert** [comment]
For SDK_RESOURCE=upstream, need to also provide UPSTREAM_JOB_NAME=build-scripts/jobs/jdk8u/jdk8u-linux-x64-hotspot and UPSTREAM_JOB_ID=1151

https://ci.adoptopenjdk.net/job/build-scripts/job/jobs/job/jdk8u/job/jdk8u-linux-x64-hotspot/1151/

**Comment by YeeSkywalker** [comment]
> For SDK_RESOURCE=upstream, need to also provide UPSTREAM_JOB_NAME=build-scripts/jobs/jdk8u/jdk8u-linux-x64-hotspot and UPSTREAM_JOB_NUMBER=1151 to the Grinder
> 
> https://ci.adoptopenjdk.net/job/build-scripts/job/jobs/job/jdk8u/job/jdk8u-linux-x64-hotspot/1151/

Thx for clarification!

File: buildenv/jenkins/JenkinsfileBase [hunk 1]
```diff
@@ -369,11 +369,12 @@ def setup() {
 
 			if (params.SDK_RESOURCE == 'nightly' && params.CUSTOMIZED_SDK_URL) {
 				CUSTOMIZED_SDK_URL_OPTION = "-c '${params.CUSTOMIZED_SDK_URL}'"
-			} else if (params.CUSTOMIZED_SDK_URL) {
-				SDK_RESOURCE = "customized"
+			}else if (params.SDK_RESOURCE != 'customized'){
+				CUSTOMIZED_SDK_URL_OPTION = ""
+			}else if (params.CUSTOMIZED_SDK_URL) {
```
Comment (smlambert) [inline buildenv/jenkins/JenkinsfileBase:374]:
```suggestion
			} else if (params.SDK_RESOURCE != 'customized'){
				CUSTOMIZED_SDK_URL_OPTION = ""
			} else if (params.CUSTOMIZED_SDK_URL) {
```

**Comment by sophia-guo** [comment]
Could you also check the following case:
| SDK_RESOURCE value  | CUSTOMIZED_SDK_URL value | JDK_IMPL | Grinder to verify |
| ------------- | ------------- | ------------- | ------------- |
| releases | "any non empty url  "  | hotspot  |  |  

which can rerun Grinder/3203 with non empty CUSTOMIZED_SDK_URL. This is the case this PR is fixing.

**Comment by smlambert** [comment]
@YeeSkywalker - can you please run the one additional Grinder that @sophia-guo suggests in https://github.com/adoptium/aqa-tests/pull/3266#issuecomment-1020379675 ?

**Comment by YeeSkywalker** [comment]
> Could you also check the following case:
> 
> SDK_RESOURCE value	CUSTOMIZED_SDK_URL value	JDK_IMPL	Grinder to verify
> releases	"any non empty url "	hotspot	
> which can rerun Grinder/3203 with non empty CUSTOMIZED_SDK_URL. This is the case this PR is fixing.

Hi Sophia, I have updagted the test table and the Grinder link to the new test is [Grinder/3294](https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/3294/console)

| SDK_RESOURCE value  | CUSTOMIZED_SDK_URL value | JDK_IMPL | Grinder to verify |
| ------------- | ------------- | ------------- | ------------- |
| nightly | ""  | hotspot  | [Grinder/3198](https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/3198/console) |  
| nightly | ""  | openj9  | [Grinder/3201](https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/3201/console) |
| nightly | Top-level link to OpenJ9 artifactory  | openj9  | [Grinder/3230](https://ci.adoptopenjdk.net/job/Grinder/3230/console) |  
| releases | ""  | hotspot  | [Grinder/3203](https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/3203/console) |  
| releases | Any non empty url  | hotspot  | [Grinder/3294](https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/3294/console) |  
| releases | ""  | openj9  | [Grinder/3207](https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/3207/console) |  
| customized  | ""  | hotspot or openj9  | [Grinder/3192](https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/3192/console) |
| customized  | A link to a binary ending in tar.gz or zip  | hotspot or openj9  | [Grinder/3226](https://ci.adoptopenjdk.net/job/Grinder/3226/console) |
| upstream  | "" | hotspot or openj9  | [Grinder/3218](https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/3218/console) |

**Comment by YeeSkywalker** [comment]
> @YeeSkywalker - can you please run the one additional Grinder that @sophia-guo suggests in [#3266 (comment)](https://github.com/adoptium/aqa-tests/pull/3266#issuecomment-1020379675) ?

Hi Shelley, I have updated the test table and the Grinder link to the new test is [Grinder/3294](https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/3294/console)

| SDK_RESOURCE value  | CUSTOMIZED_SDK_URL value | JDK_IMPL | Grinder to verify |
| ------------- | ------------- | ------------- | ------------- |
| nightly | ""  | hotspot  | [Grinder/3198](https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/3198/console) |  
| nightly | ""  | openj9  | [Grinder/3201](https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/3201/console) |
| nightly | Top-level link to OpenJ9 artifactory  | openj9  | [Grinder/3230](https://ci.adoptopenjdk.net/job/Grinder/3230/console) |  
| releases | ""  | hotspot  | [Grinder/3203](https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/3203/console) |  
| releases | Any non empty url  | hotspot  | [Grinder/3294](https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/3294/console) |  
| releases | ""  | openj9  | [Grinder/3207](https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/3207/console) |  
| customized  | ""  | hotspot or openj9  | [Grinder/3192](https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/3192/console) |
| customized  | A link to a binary ending in tar.gz or zip  | hotspot or openj9  | [Grinder/3226](https://ci.adoptopenjdk.net/job/Grinder/3226/console) |
| upstream  | "" | hotspot or openj9  | [Grinder/3218](https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/3218/console) |

**Review by sophia-guo — APPROVED** [review]
LGTM

**Review by smlambert — APPROVED** [review]
thanks @YeeSkywalker !

**Pull request merged**